src/tools/loop_metadata/src/LoopMetadata.cpp

```diff
@@ -27,27 +27,59 @@
 using namespace llvm;
 using namespace llvm::noelle;
 
-bool LoopMetadataPass::setIDs(Module &M, Noelle &noelle) {
+std::vector<LoopStructure *> LoopMetadataPass::getLoopStructuresOnly(
```

Comment:
The name is ambiguous because there are APIs called "getLoopStructures".
Understanding the difference between "getLoopStructures" and "getLoopStructuresOnly" requires looking at the implementation of these APIs.

Reply:
Also, loop_metadata is a tool, so it should relies on NOELLE's APIs rather than implementing customized ones.

Reply:
API name changed.
loop_metadata is now in src/core and does not require noelle.